src/sagemaker/model.py

```diff
@@ -407,6 +408,9 @@ def _compilation_job_config(
             else input_shape,
             "Framework": framework.upper(),
         }
+        if framework.upper() == "PYTORCH" and framework_version is not None:
```

Comment:
Can we also check for target_device is a IOC cloud target?
edge, ml_inf targets dont support framework version